extensions/amp-accordion/1.0/accordion.type.js

```diff
@@ -56,23 +57,31 @@ AccordionDef.AccordionHeaderProps;
 /**
  * @typedef {{
  *   as: (string|PreactDef.FunctionalComponent|undefined),
+ *   role: (string|undefined)
  *   className: (string|undefined),
+ *   id: (string|undefined),
  *   children: (?PreactDef.Renderable|undefined),
  * }}
  */
 AccordionDef.AccordionContentProps;
 
 /**
  * @typedef {{
- *   role: (string|undefined),
+ *   id: (string),
+ *   role: (string),
  *   onClick: (function()|undefined),
+ *   aria-controls: (string),
+ *   aria-expanded: (string),
  *   children: (?PreactDef.Renderable|undefined),
  * }}
  */
 AccordionDef.HeaderProps;
 
 /**
  * @typedef {{
+ *   id: (string),
+ *   role: (string),
+ *   aria-labelledby: (string),
```

Comment:
Ditto: private and thus not needed here.

Reply:
Per real time conversation, refactored names and added comment for non public API here: https://github.com/ampproject/amphtml/pull/31136#issuecomment-728285011